features/netsocket/EMACInterface.cpp

```diff
@@ -63,9 +61,9 @@ nsapi_error_t EMACInterface::connect()
     }
 
     return _interface->bringup(_dhcp,
-                               _ip_address[0] ? _ip_address : 0,
-                               _netmask[0] ? _netmask : 0,
-                               _gateway[0] ? _gateway : 0,
+                               _ip_address.get_ip_address() ? _ip_address.get_ip_address() : 0,
```

Comment:
Could these be shortened as:
`_ip_address ? _ip_address.get_ip_address() : 0,`

Reply:
Hmm... Even _ip_address.get_ip_address() should be enough. Michal has not done this for other interfaces like L3IP or PPP and I cannot ask him as he is on paternity leave. I will close this PR for now, The old way work as well.

Reply:
Also `bringup` should be also changed to use `SocketAddress`.